controllers/clusterurlmonitor/clusterurlmonitor_supplement.go

```diff
@@ -32,10 +32,16 @@ type ClusterUrlMonitorSupplement struct {
 	Log               logr.Logger
 	Ctx               context.Context
 	BlackBoxExporter  BlackBoxExporter
+	ClusterID         string
 }
 
 func NewSupplement(ClusterUrlMonitor v1alpha1.ClusterUrlMonitor, Client client.Client, log logr.Logger, blackBoxExporter *blackboxexporter.BlackBoxExporter) *ClusterUrlMonitorSupplement {
-	return &ClusterUrlMonitorSupplement{ClusterUrlMonitor, Client, log, context.Background(), blackBoxExporter}
+	return &ClusterUrlMonitorSupplement{
+		ClusterUrlMonitor: ClusterUrlMonitor,
+		Client:            Client,
+		Log:               log,
+		Ctx:               context.Background(),
+		BlackBoxExporter:  blackBoxExporter}
 }
 
 func (s *ClusterUrlMonitorSupplement) EnsureFinalizer() (reconcile.Result, error) {
```

Comment:
is this function the right place for the functionality? How does the ClusterID label relate to the finalizer?

Reply:
Ok thats a good question, don't know much about the finalizer. I didn't change that part. But I would assume it doesn't influence the Finalizer list? As it ClusterID is a primitive data type?

Reply:
It's more that I wouldn't want the function to set an unrelated variable as a side effect.